### PR TITLE
Low rank fix

### PR DESCRIPTION
**Pull request by lostella**
This aims at solving #31. The implemented fix is rather minimal, but I'm not very familiar with the algorithms involved, so I'm not sure this is the best way around the issue (although probably the simplest).

Tests are included that fail on 9da9aa0a0 (current master).

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -293,3 +293,19 @@ LinearAlgebra.adjoint(A::MyOp) = MyOp(adjoint(A.mat))
...
     @test svds(A, v0 = ones(9))[1].S == svds(A.mat, v0 = ones(9))[1].S
 end
+
+@testset "low rank" begin
+    @testset "$T coefficients" for T in [Float64, Complex{Float64}]
+        @testset "rank $r" for r in [2, 5, 10]
+            m, n = 3*r, 4*r
+            A = randn(T, m, r)
+            B = randn(T, r, n)
+            C = A*B
```
Comment (lostella) [inline test/runtests.jl]: This should probably be generated with known U, S, V...

**Comment by lostella** [comment]
Updated tests, now they fail due to divisions in lines 337 and 342: left singular vectors are NaN for fat matrices, right singular vectors are NaN for thin matrices. This requires further fixing.

**Comment by lostella** [comment]
Looks like in the `Complex{Float64}` case, `eigs` has a hard time producing an orthonormal basis when the provided matrix is rank deficient. This makes tests fail.

Minimal (non)working example:

```julia
julia> using Random

julia> Random.seed!(0);

julia> A = randn(Complex{Float64}, 6, 2)*randn(Complex{Float64}, 2, 8); # A is 6-by-8, rank 2

julia> using Arpack

julia> ex = eigs(A*A'; nev=4);

julia> ex[2]'ex[2]
4×4 Array{Complex{Float64},2}:
          1.0+0.0im          -2.77556e-17+2.77556e-17im  …  -5.55112e-17-8.32667e-17im
 -2.77556e-17-2.77556e-17im           1.0+0.0im              3.46945e-17-1.21431e-16im
  1.11022e-16-8.32667e-17im  -2.22045e-16+2.77556e-16im         0.158544+0.0891767im
 -5.55112e-17+8.32667e-17im   3.46945e-17+1.21431e-16im              1.0+0.0im
```

See the bottom-right corner in the matrix: not even close to being orthogonal. The above does not seem to happen in the real (`Float64`) case.

Any idea what's going on?

**Comment by andreasnoack** [comment]
Thanks for looking into this. Your fix here generally looks good to me.

I think the reason you are seeing the issue here in the complex case is that ARPACK doesn't have a Hermitian solver. Instead, it just uses the general complex solver even though the input is Hermitian. Hence the solver doesn't know that there always is a unitary basis since that is not the case for the non-Hermitian problem. I guess we could orthogonalize the vectors as a post-processing step but for this PR, I'd be fine with marking the test as broken and add a comment there about the issue.

**Pull request merged**